lib/streamlit/elements/widgets/time_widgets.py

```diff
@@ -602,6 +602,7 @@ def parse_date_deterministic(v: SingleDateValue) -> str | None:
             max_value=parsed_max_date,
             key=key,
             help=help,
+            format=format,
```

Comment:
Not sure if we want to have `format` be included in this widget ID computation since I'm fairly sure this is an argument that could be safely changed by a user without us needing to reset a widget's value since it only affects how a widget is displayed. I added it here because it's currently included in the ID computation for `st.slider`

I could see the argument for removing it from the ID hash for both widgets, though.

Reply:
For now, it seems fine to add it since (I believe) it was also part of the ID calculation before we changed to the new method.

However, there is definitely some potential to do a dedicated project and carefully remove all the parameters that are not required to be part of the ID calculation. Which would probably improve the general usability. Especially if the user has defined a `key`, it might be nice to loosen this up even more.